File: regexlib/javascript_test_file/regexlib_6148.js
/* 6148
 * <(\w+)(\s(\w*=".*?")?)*((/>)|((/*?)>.*?</\1>))
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"<1"+"\t0="""*32+"! _1_EOA(i or ii)"
 */
var REGEX = new RegExp("<(\w+)(\s(\w*=".*?")?)*((/>)|((/*?)>.*?</\1>))")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '<1' + '\t0=\"\"'.repeat(i*1) + '! _1_EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_6148.py
# 6148
# <(\w+)(\s(\w*=".*?")?)*((/>)|((/*?)>.*?</\1>))
# EXPONENT
# nums:5
# EXPONENT AttackString:"<1"+"\t0="""*32+"! _1_EOA(i or ii)"

import re2 as re
from time import perf_counter

regex = """<(\w+)(\s(\w*=".*?")?)*((/>)|((/*?)>.*?</\1>))"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "<1" + "\t0=\"\"" * i * 1 + "! _1_EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")